src/modules/ackermann_drive/AckermannDriveGuidance/AckermannDriveGuidance.cpp

```diff
@@ -0,0 +1,272 @@
+		_global_local_proj_ref.initReference(_local_position.ref_lat, _local_position.ref_lon, _local_position.ref_timestamp);
+	}
+
+	// Setup global frame variables
+	_curr_pos = {_vehicle_global_position.lat, _vehicle_global_position.lon};
+	_curr_wp = {_position_setpoint_triplet.current.lat, _position_setpoint_triplet.current.lon};
+	_next_wp = _curr_wp;
+	_prev_wp = {_home_position.lat, _home_position.lon};
+
+	if (_position_setpoint_triplet.previous.valid) {
+		_prev_wp(0) = _position_setpoint_triplet.previous.lat;
+		_prev_wp(1) = _position_setpoint_triplet.previous.lon;
+	}
+
+	if (_position_setpoint_triplet.next.valid) {
+		_next_wp(0) = _position_setpoint_triplet.next.lat;
+		_next_wp(1) = _position_setpoint_triplet.next.lon;
+	}
+
+	// Setup local frame variables
+	_curr_pos_local = {_local_position.x, _local_position.y};
+	_curr_wp_local = _global_local_proj_ref.project(_curr_wp(0), _curr_wp(1));
+	_prev_wp_local = _global_local_proj_ref.project(_prev_wp(0), _prev_wp(1));
+	_next_wp_local = _global_local_proj_ref.project(_next_wp(0), _next_wp(1));
+	updateAcceptanceRadius(_curr_wp_local, _prev_wp_local, _next_wp_local);
+
+	// Stop at final waypoint
+	if (_mission_result.finished) {
+		_currentState = GuidanceState::GOAL_REACHED;
+
+	} else {
+		_currentState = GuidanceState::DRIVING;
+	}
+
+	// Guidance logic
+	switch (_currentState) {
+	case GuidanceState::DRIVING: {
+			// Set rover speed
+			Vector2f rover_velocity = {_local_position.vx, _local_position.vy};
+			_ackermann_drive_setpoint.speed_actual = rover_velocity.norm();
+			float distance_to_prev_wp = get_distance_to_next_waypoint(_curr_pos(0), _curr_pos(1),
+						    _prev_wp(0),
+						    _prev_wp(1));
+
+			if (distance_to_prev_wp <= _prev_acc_rad) { // Cornering speed
+				float cornering_speed = _param_ad_miss_vel_tun.get() / _prev_acc_rad;
+				_desired_speed = math::constrain<float>(cornering_speed, _param_ad_miss_vel_min.get(), _param_ad_miss_vel_def.get());
+
+			} else { // Default mission speed
+				_desired_speed = _param_ad_miss_vel_def.get();
+			}
+
+			// Calculate crosstrack error
+			Vector2f prev_wp_to_curr_wp_local = _curr_wp_local - _prev_wp_local;
+			Vector2f prev_wp_to_curr_pos_local = _curr_pos_local - _prev_wp_local;
+			Vector2f distance_on_line_segment = ((prev_wp_to_curr_pos_local * prev_wp_to_curr_wp_local) /
+							     prev_wp_to_curr_wp_local.length()) * prev_wp_to_curr_wp_local.normalized();
+			Vector2f crosstrack_error = (_prev_wp_local + distance_on_line_segment) - _curr_pos_local;
+
+			// Calculate desired heading towards lookahead point
+			float desired_heading = 0.f;
+			float lookahead_distance = math::constrain<float>(_param_ad_lookahead_tun.get() * _desired_speed,
+						   _param_ad_lookahead_min.get(), _param_ad_lookahead_max.get());
+
+			if (crosstrack_error.length() > lookahead_distance) { // Crosstrack error larger than lookahead
```

Comment:
You could also use `longerThan(..)` to avoid the square root.
```suggestion
			if (crosstrack_error.longerThan(lookahead_distance)) {
```
Also remove the comment as it says the same thing as the code:
- Comment: "Crosstrack error larger than lookahead"
- Code "crosstrack_error longerThan lookahead_distance"